### PR TITLE
feat: support wire decorator

### PR DESCRIPTION
**Pull request by cardoso**
When attempting to implement #6 , I noticed the wire decorator was not available yet, so this PR aims to make `@wire` available.

I included a test component and a few wire adapters:

Screenshot 2024-06-11 at 02 50 01

There's also a test, but no mocking functionality for wire adapters yet. I think it can be done in a separate PR.

### CODE REVIEW

File: packages/@lwc/wds-core/src/node/plugins/lwc/resolve-engine.js
Comment (cardoso) [inline packages/@lwc/wds-core/src/node/plugins/lwc/resolve-engine.js:58]: This is actually the only change I had to make to enable `@wire`.

**Comment by divmain** [comment]
At first glance, this looks good to me. However, we are currently working on adding wire-mocking functionality to the test side of things. I'd like to see the final shape of that before merging your PR, in case there is anything foundational we need to change that impacts this PR.

**Comment by divmain** [comment]
We will have the wire-mocking functionality ready soon. It'll also include a similar alternative to the `patchedMeta` change that you introduced here, along with a lot of wire mocking functionality for the test runner. After that is merged, I'd like to merge the example you added to show how non-mocked wires can also work, if the wire adapter is compatible. I'll come back and merge this once the other PR is in`main`.

**Comment by cardoso** [comment]
Hi @divmain , I just synced with main. Only some extra spacing was removed from resolve-engine.js by my editor I guess.

**Review by divmain — APPROVED** [review]

**Pull request merged**
